misc/systemd/ipfs.service

```diff
@@ -0,0 +1,31 @@
+[Unit]
+Description=ipfs p2p daemon
+After=network.target
+Requires=network.target
+
+[Service]
+Type=simple
+User=ipfs
+RestartSec=1
+Restart=always
+PermissionsStartOnly=true
+Nice=18
+StateDirectory=/var/lib/ipfs
+Environment=IPFS_PATH=/var/lib/ipfs
+Environment=HOME=/var/lib/ipfs
+LimitNOFILE=8192
+Environment=IPFS_FD_MAX=8192
+EnvironmentFile=-/etc/sysconfig/ipfs
+StandardOutput=journal
+WorkingDirectory=/var/lib/ipfs
+ExecStartPre=-adduser --system --group --home /var/lib/ipfs ipfs
+ExecStartPre=-mkdir /var/lib/ipfs
```

Comment:
As with the user account creation, the systemd way to do this is to write a `tmpfiles.d` file:

```
d /var/lib/ipfs 0750 ipfs ipfs -
```